handlers/auth.ts

```diff
@@ -34,33 +34,73 @@ const addProviderSchemas = [
-  const msaId = await getMsaByPublicKey(publicKey);
-  if (!msaId || !useChallenge(challenge)) return res.status(401).send();
-  // Validate Challenge signature
-  const { isValid } = signatureVerify(challenge, encodedValue, publicKey);
-  if (!isValid) return res.status(401).send();
+export const authLogin2: Handler<T.Paths.AuthLogin2.RequestBody> = async (
+  c,
+  _req,
+  res,
+) => {
+  const { signIn, signUp } = c.request.requestBody;
+  const api = await getApi();
+  if (signUp) {
+    try {
+      const { calls, publicKey } = await validateSignup(
+        api,
+        signUp,
+        providerId,
+      );
+
+      const txns = calls?.map((x) => api.tx(x.encodedExtrinsic));
+      const callVec = api.registry.createType("Vec<Call>", txns);
+
+      await api.tx.frequencyTxPayment
+        .payWithCapacityBatchAll(callVec)
+        .signAndSend(
+          getProviderKey(),
+          { nonce: await getNonce() },
+          ({ status, dispatchError }) => {
+            if (dispatchError) {
+              console.error("ERROR in Signup: ", dispatchError.toHuman());
+            } else if (status.isInBlock || status.isFinalized) {
+              console.log("Account signup processed", status.toHuman());
+            }
+          },
+        );
+      const response: T.Paths.AuthLogin2.Responses.$200 = {
+        accessToken: await createAuthToken(publicKey),
+        expires: Date.now() + 60 * 60 * 24,
+      };
+      return res.status(200).json(response);
+    } catch (e: any) {
+      console.error(`Failed Signup validation ${e.toString()}`);
+      return res.status(401).send();
+    }
+  } else if (signIn) {
+    try {
+      const parsedSignin = await validateSignin(api, signIn, "localhost");
+      const response: T.Paths.AuthLogin2.Responses.$200 = {
+        accessToken: await createAuthToken(parsedSignin.publicKey),
+        expires: Date.now() + 60 * 60 * 24,
+      };
+      return res.status(200).json(response);
+    } catch (e) {
+      console.error(`Failed Signin: ${e}`);
+      return res.status(401).send();
+    }
+  }
 
-  const response: T.Paths.AuthLogin.Responses.$200 = {
-    accessToken: await createAuthToken(c.request.requestBody.publicKey),
-    expires: Date.now() + 60 * 60 * 24,
-    dsnpId: msaId,
-  };
-  return res.status(200).json(response);
+  // We got some bad data if we got here.
+  return res.status(500).send();
 };
 
-export const authLogout: Handler<{}> = async (_c, _req, res) => {
+export const authLogout: Handler<object> = async (_c, _req, res) => {
   return res.status(201);
 };
 
-export const authProvider: Handler<{}> = async (_c, _req, res) => {
+// TODO: Figure out a better way to do this perhaps?
+// It provides to the frontend the various direct conenctions it might need
+export const authProvider: Handler<object> = async (_c, _req, res) => {
   const response: T.Paths.AuthProvider.Responses.$200 = {
+    siwfUrl: getSiwfUrl(),
```

Comment:
is this following a convention?

Reply:
Just the one already in the codebase. Sort of a get config call